src/js/providers/flash.js

```diff
@@ -222,10 +220,8 @@ define([
                     }
                 },
                 detachMedia: function() {
```

Comment:
Can we remove `attachMedia` and `detachMedia` completely? The `_beforecompleted` logic should be handled in the model.